src/clients/ProfitClient.ts

```
@@ -367,8 +370,9 @@ export class ProfitClient {
   getFillAmountInUsd(deposit: Deposit, fillAmount: BigNumber): BigNumber {
     const l1TokenInfo = this.hubPoolClient.getTokenInfoForDeposit(deposit);
     if (!l1TokenInfo) {
+      const inputToken = sdkUtils.getDepositInputToken(deposit);
```

Note: `HubPoolClient.getTokenInfoForDeposit()` selects `originToken`.  This function should probably switch on the type of Deposit (`v2Deposit` | `v3Deposit`), and if `v3Deposit` it should use the outputToken instead.

Second headache: If this should have theoretical support for arbitrary output tokens, `outputToken` might not be represented on mainnet, so the PriceClient would need to support querying arbitrary tokens outside of mainnet. So for now I guess proceed with `inputToken` and identify this update as a potential roadmap item.

---

I think all of the changes we roll out with at first should assume output token and input are equivalent and if they are not we should throw an error or exit prematurely since we can't handle them yet

---

> This function should probably switch on the type of Deposit (v2Deposit | v3Deposit), and if v3Deposit it should use the outputToken instead.

I agree with this. Its going to ease eventual support for cross-token swaps if we grab the token info for the outputToken

As to the second headache about PriceClient not supporting `outputToken`, will need to add a `catch` statement after the `getPriceOfToken` call below?

---

For now I think we just filter the input deposit events that are fed to the ProfitClient, such that we ensure input and output tokens are equivalent. That seems simplest - otherwise we need to start making everything robust to disparate output tokens.

---

I've added a token equivalency check to the filtering that we apply on unfilled deposits. This is at home with all of the other filtering that we apply. https://github.com/across-protocol/relayer-v2/pull/1157/commits/7219e21c492c202a65c46a42a133bc0285b39348

---

Relocated to a separate PR: https://github.com/across-protocol/relayer-v2/pull/1171.